elegant/process_data.py

```diff
@@ -80,6 +85,75 @@ def annotate_stage_pos(experiment_root, position, metadata, annotations):
     annotations['stage_y'] = y
     annotations['starting_stage_z'] = z
 
+def annotate_lawn(experiment_root, position, metadata, annotations, num_images_for_lawn=3):
+    '''Position annotator used to find the lawn and associated metadata about it'''
+
+    print(f'Working on position {position}')
+    position_root = experiment_root / position
+    lawn_mask_root = experiment_root / 'derived_data' / 'lawn_masks'
+    lawn_mask_root.mkdir(parents=True, exist_ok=True)
+
+    microns_per_pixel = process_images.microns_per_pixel(metadata['objective'],metadata['optocoupler'])
+
+    position_images = load_data.scan_experiment_dir(experiment_root)[position]
+    first_imagepaths = []
+    for timepoint, timepoint_images in position_images.items():
+        if position_root / (timepoint + ' bf.png') in timepoint_images:
+            first_imagepaths.append(str(position_root / (timepoint + ' bf.png')))
+        if len(first_imagepaths) == num_images_for_lawn:
+            break
+
+    first_images = list(map(freeimage.read, first_imagepaths))
+    first_images = [process_images.pin_image_mode(image, optocoupler=metadata['optocoupler'])
+        for image in first_images]
+
+    individual_lawns = [gmm_lawn_maker(image, metadata['optocoupler']) for image in first_images]
+    lawn_mask = numpy.bitwise_or.reduce(individual_lawns, axis=0)
+
+    freeimage.write(lawn_mask.astype('uint8')*255, str(lawn_mask_root / f'{position}.png')) # Some better way to store this mask in the annotations?
+    annotations['lawn_area'] = lawn_mask.sum() * microns_per_pixel**2
+
+def gmm_lawn_maker(image, optocoupler, return_model=False):
```

Comment:
I'd say move this (and related imports) to `elegant.segment_images` to keep this file cleaner.